vaadinfrontend/src/main/java/life/qbic/datamanager/views/general/MainComponent.java

```diff
@@ -0,0 +1,43 @@
+package life.qbic.datamanager.views.general;
+
+import com.vaadin.flow.component.html.Div;
+
+/**
+ * <b>The main component</b>
+ *
+ * <p>The main component function as an abstract class containing the content and support area for
+ * individual pages in the data-manager-application</p>
+ *
+ * @since <1.0.0>
+ */
+
+public abstract class MainComponent extends Div {
+
+  private Div content;
+  private Div support;
+
+  protected MainComponent(Div content, Div support) {
+    this.content = content;
+    this.support = content;
+    content.addClassName("content");
+    support.addClassName("support");
+    add(content);
+    add(support);
+  }
+
+  public void setContent(Div content) {
```

Comment:
please provide javadoc for public methods. Otherwise extending classes do not know what is expected to happen when the method is overwritten.

Reply:
I followed your suggestion and removed this method since it's unclear if it's actually needed. I provided javadocs for the getters however :)